plugins/CoreHome/tests/UI/WidgetLoader_spec.js

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * WidgetLoader screenshot tests.
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+describe('WidgetLoader', function () {
+  this.timeout(0);
+
+  this.fixture = "Piwik\\Tests\\Fixtures\\OneVisit";
+
+  before(function () {
+      testEnvironment.testUseMockAuth = 0;
+      testEnvironment.save();
+    });
+
+  it('should redirect to the landing page when the session cookie is cleared during widget loading', async function () {
+    // We try to do an actual login
+    await page.goto("");
+    await page.type("#login_form_login", superUserLogin);
+    await page.type("#login_form_password", superUserPassword);
+    await page.evaluate(function(){
+      $('#login_form_submit').click();
+    });
+    await page.waitForNetworkIdle();
+    // check dashboard is shown
+    await page.waitForSelector('#dashboard');
+    await page.waitForNetworkIdle();
```

Comment:
Waiting for a selector is good for waiting for UI. I think we should add an `expect` assertion here, too.

```suggestion
    await page.waitForSelector('#dashboard');
    expect(await page.$('#dashboard')).to.be.ok;
```

Reply:
Thanks for the suggestion Michal!
I've added this in